### PR TITLE
Switch weakjack to submodule

### PR DESCRIPTION
**Pull request by tresf**
Switches weakjack to submodule.

Testers needed!

The AppImage (specifically, `linuxdeployqt`) was segfaulting while I was trying to test this, but it also was happening on current master, so it may be an issue with my build system.

To create an AppImage:  `make install && make appimage`

Note, this will cause a merge conflict with syncs from `stable-1.2` due to d4d909cd9e440f3eba923d748938df2064a29c4c.

### CODE REVIEW

**Comment by tresf** [comment]
AppImage for testing: https://github.com/tresf/lmms/releases/download/master/lmms-1.2.0-rc4.241-linux-x86_64.AppImage

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -67,6 +67,13 @@ ELSEIF(NOT ("${SDL_INCLUDE_DIR}" STREQUAL ""))
 	INCLUDE_DIRECTORIES("${SDL_INCLUDE_DIR}")
 ENDIF()
 
+IF(LMMS_HAVE_WEAKJACK)
+	LIST(APPEND LMMS_SRCS "${WEAKJACK_INCLUDE_DIRS}/weak_libjack.c")
```
Comment (tresf) [inline src/CMakeLists.txt]: @lukas-w I wasn't really to sure where the best place for these were, so I placed them here.   A second set of eyes is appreciated, but I think it's ready to merge.

**Comment by PhysSong** [comment]
weakjack submodule doesn't have any files matches any globs in:

https://github.com/LMMS/lmms/blob/5e2c105d5b10f59e049c2ac85264b62dd37d90b6/cmake/modules/CheckSubmodules.cmake#L21

**Comment by tresf** [comment]
@PhysSong thanks, fixed via d634ed3.  I've added `.gitignore` to the list of files to look for in a submodule.

**Pull request merged**